### ISSUE TITLE
Simple Browser and screen reader support

### ISSUE THREAD

**Issue by accesswatch** [microsoft/vscode#248456]
Type: <b>Feature Request</b>

I would like to see richer support for screen reader users in Simple Browser mode, with a specific attention to detail on selecting an element and adding it to chat. Here are a list of issues and questions for consideration:

1. I often do not see the text of the page in the virtual buffer with my screen reader. Some times I do.
2. Can we add the ability to some how select a context menu or otherwise use a hotkey if the focus can be placed on an object to copy it to chat?
3. What if one wishes to track an image or a specific blank area on the page, say a div or some other UI element? How can this be handled to supply context? Could the simple browser allow for source code view and could we highlight the code and send this to chat?

VS Code version: Code 1.100.0 (19e0f9e681ecb8e5c09d8784acaa601316ca4571, 2025-05-07T12:48:53.763Z)
OS version: Windows_NT x64 10.0.26100
Modes:

**Comment by accesswatch** [comment]
@meganrogge  do you have any suggestions on this issue?

**Comment by meganrogge** [comment]
I think adding screen reader support for the simple browser is a great idea -- @justschen do you have insight into how feasible this would be?

**Comment by justschen** [comment]
yeah this is a great idea! to address some of the points:
1. this might be because the actual content of your local hosted site has ariaLabels or doesn't have. not really sure what we could do on the vscode side to help with this, but perhaps when selecting an object, we can inject an aria-label - i can play around with this
2. working on making a keybinding for starting and cancelling selection, yes! context menu would be a bit more involved, but can be explored in future iterations 🚀 
3. using the ui element selector, there isn't really a great way at the moment for us to grab sourcemaps (afaik). when selecting with the ui element selector, it does it's best to grab the html relevant to the element + the css. with screenshotting, we could introduce a built in screenshot tool that let's you pick where to screenshot though.

all great feedback! thanks!

**Comment by accesswatch** [comment]
I think at a minimum there should be a way to do this in source view of some kind @meganrogge do you have any feedbback on this one? While I can set context here with code getting the visual/images would also be useful here after highlighting the source code/html of the page.

**Comment by accesswatch** [comment]
Also, one more comment here, when using the simple browser, can we show the page in VS Code like we do for previews of Markdown or Live Share?

**Comment by meganrogge** [comment]
@accesswatch what do you mean by a source view?

**Comment by meganrogge** [comment]
Yes, @justschen and I have discussed that we should make this keyboardn navigable and it will be in the works.
